plugins/eos-blacklist/gs-plugin-eos-blacklist.c

```diff
@@ -207,20 +215,48 @@ app_get_flatpak_ref_name (GsApp *app)
 	return gs_app_get_metadata_item (app, "flatpak::RefName");
 }
 
+static gchar **
+get_default_locales (GsPlugin *plugin)
+{
+	g_auto(GStrv) flatpak_default_locales = NULL;
+	g_autoptr(GError) local_error = NULL;
+
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	if (priv->installation == NULL)
+		return NULL;
+
+	flatpak_default_locales = flatpak_installation_get_default_languages (priv->installation, &local_error);
+	if (local_error != NULL) {
+		g_warning ("No user locales could be retrieved! %s", local_error->message);
+		return NULL;
+	}
+
+	return g_steal_pointer (&flatpak_default_locales);
+}
+
 static gboolean
 gs_plugin_locale_is_compatible (GsPlugin *plugin,
 				const char *locale)
 {
 	g_auto(GStrv) locale_variants;
+	g_auto(GStrv) locale_options = NULL;
 	const char *plugin_locale = gs_plugin_get_locale (plugin);
 	int idx;
 
+        /* Check if a variant of a locale is compatible, e.g. 'pt' for 'pt_BR' */
 	locale_variants = g_get_locale_variants (plugin_locale);
 	for (idx = 0; locale_variants[idx] != NULL; idx++) {
 		if (g_strcmp0 (locale_variants[idx], locale) == 0)
 			return TRUE;
 	}
 
+        /* check if the app's locale is listed in the languages key for the ostree repo file */
+	locale_options = get_default_locales (plugin);
+	for (idx = 0; locale_options != NULL && locale_options[idx] != NULL; idx++) {
+		if (g_strcmp0 (locale_options[idx], locale) == 0)
+			return TRUE;
+	}
```

Comment:
You could actually simplify this further using [`g_strv_contains()`](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strv-contains):
```
locale_options = get_default_locales (plugin);
if (locale_options != NULL && g_strv_contains (locale_options, locale))
	return TRUE;
```

Sorry I didn’t think of this before. You could also simplify the loop above using `g_strv_contains()` (as a separate commit).

Reply:
Oh yeah, that makes sense :) will do!